File: crm/src/verticals/food.ts
import { tab } from './shared';
import type { CrmVerticalDefinition } from './types';

export const foodVertical: CrmVerticalDefinition = {
  key: 'food',
  label: 'Food',
  riskLevel: 'low',
  entityLabelSingular: 'menu item',
  entityLabelPlural: 'menu items',
  defaultPlanLabel: 'Food ordering plan',
  clientTabs: [
    tab('overview', 'Overview'),
    tab('readiness', 'Readiness'),
    tab('catalog', 'Menu'),
    tab('crawl', 'Sources'),
    tab('leads', 'Orders/Leads'),
    tab('activity', 'Conversations'),
    tab('prompt', 'Prompt'),
    tab('controls', 'Controls'),
  ],
  entityTypes: ['restaurant', 'menu_item', 'grocery_item', 'cuisine', 'offer', 'delivery_zone'],
  readinessChecks: ['menu', 'location', 'delivery_zone', 'cart', 'checkout', 'dietary_data'],
};

File: agent/verticals/food.py
"""Food and grocery vertical definition."""

from agent.verticals.base import VerticalDefinition, tabs

VERTICAL = VerticalDefinition(
    key="food",
    label="Food",
    risk_level="low",
    entity_label_singular="menu item",
    entity_label_plural="menu items",
    default_plan_label="Food ordering plan",
    crm_tabs=tabs(
        ("overview", "Overview"),
        ("readiness", "Readiness"),
        ("catalog", "Menu"),
        ("crawl", "Sources"),
        ("leads", "Orders/Leads"),
        ("activity", "Conversations"),
        ("prompt", "Prompt"),
        ("controls", "Controls"),
    ),
    entity_types=("restaurant", "menu_item", "grocery_item", "cuisine", "offer", "delivery_zone"),
    readiness_checks=("menu", "location", "delivery_zone", "cart", "checkout", "dietary_data"),
    action_types=("SHOW_ENTITIES", "SET_LOCATION", "ADD_TO_CART", "CHECKOUT_HANDOFF", "CAPTURE_LEAD"),
)
